### DISCUSSION TITLE
Granular withoutEvents trait

### DISCUSSION THREAD

**Discussion by vicenterusso** [laravel/framework#51181] [Ideas]
Should withoutEvents receive parameters to ignore only certain events?

```php
$user = User::withoutEvents(function () {
    User::findOrFail(1)->delete();
 
    return User::find(2);
});
```

Would be something like this

```php
$user = User::withoutEvents(['saved'], function () {
    User::findOrFail(1)->delete();
    return User::find(2);
});
```

So, in this case, only `saved` event is skipped.

**Comment by henzeb** [comment]
If you want to do that, it would be the second parameter, or you are willing to wait a year? It's breaking change this way.

**Reply by vicenterusso** [reply]
yeah, could be second parameter.. it was just an idea
